graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/QueryValidationResource.java

```diff
@@ -138,8 +143,18 @@ private ValidationRequest prepareRequest(ValidationRequestDTO validationRequest,
     }
 
     private Set<ExplainResults.IndexRangeResult> indexRanges(ValidationRequest request) {
+        final Set<String> streamTitles = request.streams().stream()
+                .map(streamId -> {
+                    try {
+                        return streamService.load(streamId).getTitle();
```

Comment:
This could be a performance issue. If the user does not specify a stream, we would call this for all streams and the `/search/validate` endpoint is also called on every keystroke.

Reply:
Added stream title caching